clion/src/main/kotlin/org/rust/clion/debugger/RsCLionDebuggerDriverConfigurationProvider.kt

```diff
@@ -15,6 +17,14 @@ import org.rust.debugger.RsDebuggerDriverConfigurationProvider
 class RsCLionDebuggerDriverConfigurationProvider : RsDebuggerDriverConfigurationProvider {
     override fun getDebuggerDriverConfiguration(project: Project, isElevated: Boolean): DebuggerDriverConfiguration? {
         val toolchain = CPPToolchains.getInstance().defaultToolchain ?: return null
-        return createDriverConfiguration(project, CPPEnvironment(toolchain), isElevated)
+        val isLLDBRustMSVCSupportEnabled = Registry.`is`("org.rust.debugger.lldb.rust.msvc", false)
+
+        return if (toolchain.toolSet.isMSVC && isLLDBRustMSVCSupportEnabled) {
+            object : CLionLLDBDriverConfiguration(project, CPPEnvironment(toolchain), isElevated) {
+                override fun useRustTypeSystem(): Boolean = true
+            }
```

Comment:
I'm slightly worried about the creation of concrete configuration manually here because if something changes `createDriverConfiguration`, our code won't do the same thing automatically. But I think it's ok for now